core/src/executor/evaluate/error.rs

```diff
@@ -100,6 +100,20 @@ pub enum EvaluateError {
 
     #[error("unsupported evaluate string unary factorial: {0}")]
     UnsupportedUnaryFactorial(String),
+
+    #[error("unsupported custom function in subqueries")]
+    UnsupportedCustomFunction,
+
+    #[error("function args.length not matching: {name}, expected: {expected_minimum} ~ {expected_maximum}, found: {found}")]
+    FunctionArgsLengthNotWithinRange {
+        name: String,
+        expected_minimum: usize,
+        expected_maximum: usize,
```

Comment:
@ever0de current custom function does not support open range so it would be better to use `expected_minimum` & `expected_maximum`.
`Range` can describe `(3..)` or `(..=4)` like open ranges that are not supported by our custom function args